### PR TITLE
Feature/issue 707 ops partials mv 2

### PR DESCRIPTION
**Pull request by wds15**
#### Submission Checklist

- [ ] Run unit tests: `./runTests.py test/unit` (I ran the operations_and_partials tests)
- [X] Run cpplint: `make cpplint`
- [X] Declare copyright holder and open-source license: see below

#### Summary:
Correct `operands_and_partials` to work for nested containers as designed. This is needed for multi-variate use of `operands_and_partials`.

#### Intended Effect:
Enable use of `operands_and_partials` with multi-variate distributions which rely on the definition of `partials_vec_` as being part of the edges. The current implementation did not have a `partials_vec_` if an edge was given as `double` such that the usual code pattern would not compile. Now adequate specializations are added to handle the mvn cases correctly.

#### How to Verify:
A test at the end of `./test/unit/math/rev/mat/meta/operands_and_partials_test.cpp` was added. The test is essentially fulfilled if the test compiles. On current develop the test will fail to compile even though this is the intended use case of the `operands_and_partials`

#### Side Effects:
When an edge is a nested container type, the the size of the overall empty object increases from 5 to 6.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Sebastian Weber

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by wds15** [comment]
I have no idea what Jenkins has to complain about... and travis failed on my other pull with some errors on packages not being available...

**Comment by seantalts** [comment]
In the [console output](http://d1m1s1b1.stat.columbia.edu:8080/job/Math%20Pipeline/job/PR-711/1/consoleFull) there's a little more info. During `make test-headers` it found that I didn't include some necessary stuff in the new `prim/mat..operands_and_partials`. I can fix it on your branch.

File: stan/math/prim/scal/meta/operands_and_partials.hpp [hunk 1]
```diff
@@ -28,10 +28,11 @@ namespace internal {
 template <typename ViewElt, typename Op>
 class ops_partials_edge {
  public:
-  empty_broadcast_array<ViewElt, Op> partials_;
+  typedef empty_broadcast_array<ViewElt, Op> partials_t;
+  partials_t partials_;
```
Comment (seantalts) [inline stan/math/prim/scal/meta/operands_and_partials.hpp]: Oops, I think we can actually revert the changes on this file - we don't need the `typedef` anymore and we typically add a commented out argument name to make it clear that we don't use the argument (the `/* op */` below)
